docs/troubleshooting/networking-and-proxies.mdx

```diff
@@ -0,0 +1,217 @@
+
+### Basic Configuration
+
+**Windows (Command Prompt)**
+```cmd
+set https_proxy=http://proxy.company.com:8080
+set http_proxy=http://proxy.company.com:8080
+cline start
+```
+
+**Windows (PowerShell)**
+```powershell
+$env:https_proxy="http://proxy.company.com:8080"
+$env:http_proxy="http://proxy.company.com:8080"
+cline start
+```
+
+**macOS/Linux**
+```bash
+export https_proxy=http://proxy.company.com:8080
+export http_proxy=http://proxy.company.com:8080
+cline start
+```
+
+### Proxy with Authentication
+
+If your proxy requires authentication, include credentials in the URL:
+
+```bash
+export https_proxy=http://username:password@proxy.company.com:8080
+export http_proxy=http://username:password@proxy.company.com:8080
+```
+
+<Warning>
+Storing credentials in environment variables can be a security risk.
+</Warning>
+
+### Bypass Proxy for Localhost
+
+To prevent localhost traffic from going through the proxy, set the `no_proxy` environment variable:
+
+**Windows**
+```cmd
+set no_proxy=localhost,127.0.0.1,.local
+```
+
+**macOS/Linux**
+```bash
+export no_proxy=localhost,127.0.0.1,.local
+```
+
+### Custom Certificate Authority
+
+If your proxy uses a custom CA certificate:
+
+**Windows**
+```cmd
+set NODE_EXTRA_CA_CERTS=C:\path\to\ca-certificate.crt
+cline start
+```
+
+**macOS/Linux**
+```bash
+export NODE_EXTRA_CA_CERTS=/path/to/ca-certificate.pem
+cline start
+```
+
+### Permanent Configuration
+
+To avoid setting these variables every time, add them to your shell profile or system environment variables.
+
+**macOS/Linux** (add to `~/.bashrc`, `~/.zshrc`, or `~/.profile`):
+```bash
+# Proxy configuration
+export https_proxy=http://proxy.company.com:8080
+export http_proxy=http://proxy.company.com:8080
+export no_proxy=localhost,127.0.0.1,.local
+export NODE_EXTRA_CA_CERTS=/path/to/ca-certificate.pem
+```
+
+**Windows** (System Environment Variables):
+1. Search for "Environment Variables" in Windows Settings
+2. Add the variables under "User variables" or "System variables"
+3. Restart your terminal or IDE
+
+### Known Limitations
+
+Cline CLI only supports HTTP proxies. It does not support SOCKS proxies,
+proxy autoconfiguration (PAC) scripts, or HTTP proxies which require
+authentication beyond a basic username and password.
+
+## JetBrains IDEs
+
+The JetBrains plugin uses the IDE's HTTP proxy settings.
+
+### Configure JetBrains Proxy
+
+1. Open Settings/Preferences:
+   - **Windows/Linux**: File > Settings
+   - **macOS**: IntelliJ IDEA > Preferences
+   - Or press `Ctrl+Alt+S` (Windows/Linux) or `Cmd+,` (macOS)
+
+2. Navigate to:
+   ```
+   Appearance & Behavior > System Settings > HTTP Proxy
+   ```
+
+3. Select "Manual proxy configuration"
+
+4. Configure your proxy:
+   - **Host name**: `proxy.company.com`
+   - **Port number**: `8080`
+   - Check "Proxy authentication" if required
+   - Enter your username and password
+
+5. Click "Check connection" to verify the settings
```

Comment:
In my previous testing I mentioned the caveat 

>Once I set the http proxy in Jetbrains I had to close the current window and then click on the project open thing and then it would work. If I didn't do that jetbrains would not use the proxy that I had set. Again maybe this is not something we can solve or maybe this is just a debugger thing but it took me a bit to figure that out.


It might be worthy to mention this as a Caveat either here or in trouble shooting